ocean_lib/ocean/util.py

```diff
@@ -85,7 +85,7 @@ def get_address_of_type(
         if not isinstance(addresses[address_type], dict)
         else addresses[address_type].get(key, addresses[address_type]["1"])
     )
-    return Web3.toChecksumAddress(address)
+    return Web3.toChecksumAddress(address.lower())
```

Comment:
I think toChecksumAddress already performs lower. Can you try it in a python console and remove the `lower()` part if I'm right about that?

Reply:
please check unresolved comments, like this one and the one below.

Reply:
You can check https://github.com/singnet/snet-cli/pull/142#discussion_r805788133. You need to provide a lowercase address then call web3.toChecksumAddress()
Error message is more descriptive: `web3.exceptions.InvalidAddress: ('Web3.py only accepts checksum addresses. The software that gave you this non-checksum address should be considered unsafe, please file it as a bug on their platform. Try using an ENS name instead. Or, if
 you must accept lower safety, use Web3.toChecksumAddress(lower_case_address).`

Reply:
thank you, Maria. Can you please leave a comment similar to this one, but somewhere in the code? Wherever you find it appropriate.

Reply:
Maybe in the `_checksum_addresses` function, I think it fits there.